the first delivery

Elizaveta-pleshkova opened the pull request.

Elizaveta-pl committed:
add allure report

Elizaveta-pl committed:
the first delivery

pomegranate282922 commented on src/main/java/ru/praktikum/model/OrderCreateReqest.java:
```diff
@@ -0,0 +1,104 @@
+
+    public void setColor(List<String> color) {
+        this.color = color;
+    }
```
Можно применить библиотеку lombok для уменьшения количество boilerplate кода 
Можно улучшить

pomegranate282922 commented on src/main/java/ru/praktikum/steps/CourierSteps.java:
```diff
@@ -0,0 +1,50 @@
+                .and()
+                .body(courierCreateRequest)
+                .when()
+                .post("/api/v1/courier")
```
Все урлы надо вынести как константы 
Нужно исправить

pomegranate282922 commented on src/main/java/ru/praktikum/steps/CourierSteps.java:
```diff
@@ -0,0 +1,50 @@
+                .then();
+    }
+
+    public ValidatableResponse loginCourier(String login, String password){
```
Нужно расставить степы с описаниями 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/praktikum/AddCourierTest.java:
```diff
@@ -0,0 +1,85 @@
+
+    @Before
+    public void setUp() {
+        RestAssured.baseURI = "https://qa-scooter.praktikum-services.ru";
```
Base uri присваиваем один раз в проекте. Можно использовать request spec в RestAssured или класс предок 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/praktikum/AddCourierTest.java:
```diff
@@ -0,0 +1,85 @@
+    public void setUp() {
+        RestAssured.baseURI = "https://qa-scooter.praktikum-services.ru";
+
+        login = RandomStringUtils.randomAlphabetic(10);
```
Отлично что есть рандомизация

pomegranate282922 commented on src/test/java/ru/praktikum/AddCourierTest.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    @Test
+    public void createNewCourierNotLogin_ShouldReturnIncorrectResult400(){
```
Всем тестам надо дать аллюр описания 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/praktikum/AddCourierTest.java:
```diff
@@ -0,0 +1,85 @@
+
+        courierSteps
+                .createCourier(null, null, firstName)
+                .statusCode(400);
```
Нужно везде проверить не только код ответа, но и тело ответа 
Нужно исправить

pomegranate282922 commented on src/test/java/ru/praktikum/GettingListOrdersTest.java:
```diff
@@ -0,0 +1,25 @@
+        orderSteps
+                .getOrder()
+                .statusCode(200)
+                .body("orders", CoreMatchers.notNullValue());
```
Лучше проверить непустоту списка 
Можно улучшить

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Есть рандомизация. Везде есть удаление данных. 
замечания:
1. Урлы вынести как константы 
2. Расставить степы с именованием 
3. Дать аллюр описания тестам 
4. Добавить проверки для тела ответа 
5. Остальное по коду

Elizaveta-pl committed:
add allure report

Elizaveta-pl committed:
Urls and baseUri are rendered as constants
There are steps with naming
Added gait descriptions to tests
Added checks for the response body

pomegranate282922 commented on src/main/java/ru/praktikum/config/RestConfig.java:
```diff
@@ -0,0 +1,10 @@
+
+import lombok.experimental.UtilityClass;
+
+import javax.swing.text.Utilities;
```
Лишние импорты лучше убирать

pomegranate282922 commented on src/test/java/ru/praktikum/GettingListOrdersTest.java:
```diff
@@ -0,0 +1,30 @@
+    @Before
+    @Step("Подготовка к тестам. Описание BaseURI.")
+    public void setUp() {
+        RestAssured.baseURI = RestConfig.HOST;
```
Вот это лучше вынести в класс предок

pomegranate282922 reviewed (approved):
Пара замечаний осталась, но будем считать некритичным. Зачет!
